src/Servers/Kestrel/Transport.MsQuic/Internal/QuicRegistration.cs

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal
+{
+    internal class QuicRegistration : IDisposable
```

Comment:
I think naming this class QuicRegistration is too confusing. This maps directly to the `QUIC_API_V1` struct. It'd be better to call this at least QuicApi.